stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/channel/ChannelController.kt

```diff
@@ -371,7 +377,11 @@ public class ChannelController internal constructor(
     }
 
     internal suspend fun leave(): Result<Unit> {
-        val result = channelClient.removeMembers(domainImpl.currentUser.id).execute()
+        val currentUser = domainImpl.user.value
+
+        val result = currentUser?.let { user ->
+            channelClient.removeMembers(user.id).execute()
+        } ?: Result(ChatError("Current user is not set, then it is not possible to leave channel. Please set User for ChatDomain."))
```

Comment:
Should we raise error here?

Reply:
🤔  Instead of returning the result with error? But wouldn't a crash be worse?